rbc/src/main/java/com/library/rbc/service/BookService.java

```diff
@@ -108,12 +104,21 @@ public String uploadImage(MultipartFile image){
       if (!Files.exists(uploadPath)) {
         Files.createDirectories(uploadPath);
       }
-      InputStream inputStream = image.getInputStream();
-      Path filePath = uploadPath.resolve(Objects.requireNonNull(image.getOriginalFilename()));
-      Files.copy(inputStream, filePath, StandardCopyOption.REPLACE_EXISTING);
+      String originalFilename = Objects.requireNonNull(image.getOriginalFilename());
+      String extension = getFileExtension(originalFilename);
+      String uniqueFilename = UUID.randomUUID() + extension;
```

Comment:
We should store the unique file name of the image with the other information about the associated book.